### PR TITLE
Revise annotation setup

### PR DESCRIPTION
**Pull request by hcorson-dosch-usgs**
Alright! I think I've got the new functionality in place.

### Changes to annotation set up
* On desktop, there is an annotation container at the bottom of the page, just like on mobile
  * This is populated based on the contents of the newly created `src/assets/text/droughtNarrations_desktop.js`
  * As you scroll, these are triggered by invisible rectangles that are added to the dynamic svg, the placement and height of which is determined from the `start_date` and `end_date` parameters for each narration
* On desktop, the annotations that are added directly to the chart are now in `src/assets/text/droughtAnnotations_desktop.js`
  * As before, these are placed based on the attribute fields for each annotation, and act as their own trigger upon scroll
  * There is now the option to specify that an annotation is a quote, which means the annotation is styled differently and links to the source of the quote
* On mobile, as before, all text appears in the annotation container at the bottom of the page
  * This is populated based on the contents of the newly created `src/assets/text/droughtAnnotations_mobile.js`
  * As before, the `src/assets/text/droughtAnnotations_mobile.js` data is also used to add circles to the timeline - these are placed based on the attribute fields for each annotation
  * As before, as you scroll, the appearance of the text and highlighting of the associated circle are triggered by invisible rectangles that are added to the dynamic svg, the placement and height of which is determined based on the `date` parameter for the annotation
* On desktop and on mobile, the text in the annotation-container is added dynamically with `v-html`, so that links can be included as desired.
* On desktop and on mobile, images are placed on top of the chart based on the content of `src/assets/text/droughtImages.js`
  * The images are placed with D3, based on the `date` field and the desktop- and mobile-specific placement fields
  * The images act as their own trigger on scroll

### Additional changes
* Add intro section and update title, per Xd mockup
* Add placeholders for later sections of page
* On desktop, add static titles to the grey drought event scroll-to rectangles
* Place annotation container in `#grid-container` div (though not _in_ grid) and make sticky, so that it scrolls up with the grid when the user scrolls to later sections
* On desktop, hide the annotation container on page load, and have it appear when the user begins to scroll down the page

### To test
Pull the changes to your local clone and run `npm run serve`.

### Notes
* All annotations, narrations, and images are placeholders for now. 
* The placement of all scroll triggers will likely need to be adjusted as the content is finalized.
* The images are currently triggered independently -- if we end up wanting them to be triggered alongside text or narrations I can make that switch
* The same date is currently used to place the images on the timeline on desktop and mobile -- I can modify that approach if needed, so that different dates can be used for the two views
* We could elect to place the annotation container _in_ the grid. I've left that css in, commented out. This means it would overlap the chart at the end, blocking the end of the timeline. I think if we wanted that look, we'd want to transition the annotation container so that it disappeared/shrank when the end of the timeline got near the top of the page. I started to explore that but paused b/c it seemed like for now it made sense to retain the option to have annotations all the way to the end of the timeline. We can revisit later if needed.

### CODE REVIEW

**Comment by hcorson-dosch-usgs** [comment]
Marking as draft b/c something funky is going on with the mobile preview and I can't figure it out

**Comment by hcorson-dosch-usgs** [comment]
K figured it out - the new svgs were messing up the display width

File: src/components/DroughtHistory.vue [hunk 1]
```diff
@@ -608,21 +813,71 @@ $writeFont: 'Nanum Pen Script', cursive;
...
 .droughtText.mobile {
   z-index: 10;
   font-weight: 500;
   font-size: 1em;
 }
+.droughtText.quote {
+  font-style: italic;
+}
+.droughtText.desktop.quote {
+  text-shadow: 2px 4px 4px rgba(179,179,179,0.6);
```
Comment (aaarcher-usgs) [inline src/components/DroughtHistory.vue:872]: wondering if a small neat line at bottom of the text box might help in this case, where it scrolls to next section? 

image

**Review by aaarcher-usgs — APPROVED** [review]
This is really looking sharp. 

On Safari Desktop:

image

image

image


My biggest concern is why Firefox mobile isn't picking up the changes?

File: src/components/DroughtHistory.vue [hunk 2]
```diff
@@ -32,29 +40,76 @@
...
+      class="page-section"
+    >
+      <h3>References</h3>
+      <p>These are the references</p>
+    </section>
+    <section
+      id="authors-container"
+      class="page-section"
+    >
+      <h3>Authors</h3>
```
[the remark this answers is not in the document]
Comment (hcorson-dosch-usgs) [reply src/components/DroughtHistory.vue:110]: That was my thought, yes

File: src/components/DroughtHistory.vue [hunk 3]
```diff
@@ -77,7 +132,10 @@ import { isMobile } from 'mobile-device-detect';
 import { ScrollTrigger } from "gsap/ScrollTrigger"; // animated scroll events
 import { TimelineMax } from "gsap/all";
 import { ScrollToPlugin } from "gsap/ScrollToPlugin";
-import droughtAnnotations from "@/assets/text/droughtAnnotations.js";
+import droughtAnnotationsDesktop from "@/assets/text/droughtAnnotations_desktop.js";
+import droughtAnnotationsMobile from "@/assets/text/droughtAnnotations_mobile.js";
+import droughtNarrations_desktop from "@/assets/text/droughtNarrations_desktop.js";
```
[the remark this answers is not in the document]
Comment (hcorson-dosch-usgs) [reply src/components/DroughtHistory.vue:137]: No... the terminology got a bit confusing here b/c I was trying to retain some of the original setup, which used 'annotations' throughout. Narrations are new and are only used on desktop - it's just the text content that goes in the `annotation-container` on desktop. The text content on the chart itself is in `droughtAnnotations_desktop.js``. On mobile, all text content is in the `annotation-container`, so everything is in `droughtAnnotations_mobile.js`

File: src/components/DroughtHistory.vue [hunk 4]
```diff
@@ -77,7 +132,10 @@ import { isMobile } from 'mobile-device-detect';
 import { ScrollTrigger } from "gsap/ScrollTrigger"; // animated scroll events
 import { TimelineMax } from "gsap/all";
 import { ScrollToPlugin } from "gsap/ScrollToPlugin";
-import droughtAnnotations from "@/assets/text/droughtAnnotations.js";
+import droughtAnnotationsDesktop from "@/assets/text/droughtAnnotations_desktop.js";
```
[the remark this answers is not in the document]
Comment (hcorson-dosch-usgs) [reply src/components/DroughtHistory.vue:135]: Yes, agreed. That's a downside to this approach, but I think it keeps these `.js` files cleaner and implementation cleaner.

File: src/components/DroughtHistory.vue [hunk 5]
```diff
@@ -608,21 +813,71 @@ $writeFont: 'Nanum Pen Script', cursive;
...
+.page-section {
+  margin: auto;
+  max-width: 90vw;
+}
+#region-grid-container {
+  max-width: 95vw;
+  display: grid;
+  grid-template-columns: 100%;
+  grid-template-rows: max-content max-content;
+  grid-template-areas:
```
[the remark this answers is not in the document]
Comment (hcorson-dosch-usgs) [reply src/components/DroughtHistory.vue:825]: Yes this was a placeholder for now, but will become a grid setup that differs between desktop and mobile

File: src/components/DroughtHistory.vue (hunk 1 above)
Comment (hcorson-dosch-usgs) [reply src/components/DroughtHistory.vue:872]: Yeah I think that does help:
image

File: src/assets/text/droughtAnnotations_desktop.js [hunk 6]
```diff
@@ -2,18 +2,18 @@ export default {
     timelineEvents: [
       {
         id: '1', // Unique id for annotation - order and # value don't matter - annotations are later sorted by date
-        date: '1921-10-01', // Date marking start of annotation. Determines y placement on deskop and mobile AND scroll order for annotations on mobile
-        text: 'Event 1: 10/1/1921', // Text of annotation
-        mobile_x_offset_per: 20, // x placement on mobile, as percent of chart width (0 = left edge, 100 = right edge)
```
[the remark this answers is not in the document]
Comment (hcorson-dosch-usgs) [reply src/assets/text/droughtAnnotations_desktop.js:7]: Hmm looks fine on my Firefox mobile preview:
image

We can do a better check once we push to beta

**Comment by hcorson-dosch-usgs** [comment]
Thanks for the review!! Going to push to beta now

**Pull request merged**